pkg/workload/workload.go

```diff
@@ -604,6 +604,26 @@ func QueuedWaitTime(wl *kueue.Workload, clock clock.Clock) time.Duration {
 	return clock.Since(queuedTime)
 }
 
+// PodsReadyToPreemptedTime is the amount of time it takes a workload's pods running to getting preempted.
+// This measures runtime of workloads that do not run to completion (ie are preempted).
+func PodsReadyToPreemptedTime(wl *kueue.Workload) *time.Duration {
+	var podsReady *time.Time
+	var preempted *time.Time
+	if c := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadPodsReady); c != nil {
+		if c.Status == metav1.ConditionTrue {
+			podsReady = &c.LastTransitionTime.Time
+		}
+	}
+	if c := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadPreempted); c != nil {
```

Comment:
As suggest in the other comments, and by the metric name, let's check `WorkloadEvicted` condition, and retrieve its reason too.

Reply:
Whats the difference between eviction and preemption?

Reply:
lgtm, yes we don't need to actually retrieve (return) the reason from this function